framework/source/class/qx/ui/window/Window.js

```diff
@@ -632,17 +679,34 @@ qx.Class.define("qx.ui.window.Window",
         this.hide();
         this.fireEvent("close");
       }
+
+      // If automatically destroying the window upon close was requested, do
+      // so now. (Note that we explicitly re-obtain the autoDestroy property
+      // value, allowing the user's close handler to enable/disable it before
+      // here.)
+      if (this.getAutoDestroy()) {
+        this.removeAll();                                    // remove contents
+        qx.event.Registration.removeAllListeners(this);      // remove listeners
+        qx.core.Init.getApplication().getRoot().remove(this);// remove self
+        this.dispose();
```

Comment:
Can this be reduced to just `this.dispose()`?  `this.removeAll()` is done by the `Widget.destruct` already, and `qx.event.Registration.removeAllListeners` is handled by `Object.destruct`.  

I would expect that `getRoot().remove()` does need to be called, but in the current implementation it looks like it's *never* called, even if the Window is manually destroyed -- which would sound like a bug you'd expect to have cropped up by now, so maybe it's already handled elsewhere somehow?

However, if something needs to be done during destruction/disposal, the code could be moved into the destructor or overridden `dispose()` method so it's always implemented